sdk/core/azure-core/src/main/java/com/azure/android/core/util/paging/PagedDataCollection.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package com.azure.android.core.util.paging;
+
+import androidx.annotation.NonNull;
+
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Objects;
+
+/**
+ * Represents a collection of paged data with fixed page size where pages can be retrieved synchronously
+ * @param <T> type of the items contained in the page
+ * @param <P> page that contains the items
+ */
+public class PagedDataCollection<T, P extends Page<T>> {
+    private final PagedDataRetriever<T, P> pagedDataRetriever;
+    private LinkedHashMap<String, P> pages = new LinkedHashMap<String, P>();
+    private String firstPageId;
+
+    /**
+     * Constructor requires a synchronous paged data provider
+     * @param pagedDataRetriever paged data provider with fixed page size
+     */
+    public PagedDataCollection(@NonNull PagedDataRetriever<T, P> pagedDataRetriever) {
+        Objects.requireNonNull(pagedDataRetriever);
+        this.pagedDataRetriever = pagedDataRetriever;
+    }
+
+    /**
+     * Retrieve the first page in the collection
+     * @return First page of the collection
+     */
+    public P getFirstPage() {
+        if (firstPageId != null) {
+            return pages.get(firstPageId);
+        }
+        P firstPage = pagedDataRetriever.getFirstPage();
+        firstPageId = firstPage.getPageId();
+        pages.put(firstPage.getPageId(), firstPage);
```

Comment:
nit; We can use the value we just set.
```suggestion
        pages.put(firstPageId, firstPage);
```